lib/Sema/CSSimplify.cpp

```diff
@@ -1864,6 +1864,23 @@ assessRequirementFailureImpact(ConstraintSystem &cs, Type requirementType,
     if (!cs.findSelectedOverloadFor(calleeLoc))
       return 10;
   }
+  
+  auto resolvedTy = cs.simplifyType(requirementType);
+
+  // Increase the impact of a conformance fix for generic parameters on
+  // operators where such conformance failures are not as important as argument
+  // mismatches or contextual failures.
+  if (isExpr<OverloadedDeclRefExpr>(anchor) &&
+      locator.isForRequirement(RequirementKind::Conformance) &&
+      resolvedTy->is<ArchetypeType>()) {
+    auto *calleeLoc = cs.getCalleeLocator(cs.getConstraintLocator(locator));
+    if (auto overload = cs.findSelectedOverloadFor(calleeLoc)) {
+      const auto *decl = overload->choice.getDeclOrNull();
+      if (decl && decl->isOperator()) {
```

Comment:
`OverloadedDeclRefExpr` actually contains all the information you need here, because it only contains one kind of declaration, so you can simply get `front()` and check whether it's an operator or not. Maybe even worth adding a separate method on `OverloadedDeclRefExpr` itself...

Reply:
Ah great, it's much simpler
I'll fix it, thanks :)

Reply:
No worries! :)